src/codegen/themesParser.ts

```diff
@@ -0,0 +1,124 @@
+const CSS_COMMENT_END_MATCH = new RegExp(/\*\//, 'gi');
+
+interface PropertySection {
+    [property: string]: string;
+}
+
+export interface ThemeDefinition {
+    [section: string]: PropertySection;
+}
+
+export interface ThemesData {
+    [theme: string]: ThemeDefinition;
+}
+
+export default class ThemesParser {
+    constructor() { }
+
+    public async readThemeFiles() {
+        let globalVariables = {};
+        const globalColors = await ThemesParser.parseCSSFile(COLORS_PATH); // Colors file has the global colors that are shared across themes
+        for (const globalSection of Object.values(globalColors)) {
+            globalVariables = {
+                ...globalSection,
+                ...globalVariables
+            };
+        }
+
+        const themesData: ThemesData = {};
+        const files = await promisify(readdir)(THEMES_DIR);
+        for (const filename of files) {
+            const filenameMatch = filename.match(THEME_FILENAME_PATTERN);
+            if (filenameMatch) {
+                const themeName = filenameMatch[filenameMatch.length - 1];
+                const themeFileData = await ThemesParser.parseCSSFile(join(THEMES_DIR, filenameMatch[0]), globalVariables);
+
+                themesData[themeName] = themeFileData;
+            }
+        }
+
+        return themesData;
+    }
+
+    /**
+     * Parses a css theme file.
+     * 
+     * Theme files all attach to the following rules: 
+     * - The file name indicates the theme name (_color.${themeName}.css), 
+     *   this same one matches the root attribute definition
+     * - Each section starts with a one line comment that indicates the section name
+     *   \/\* Section Name \*\/
+     * - There's no variables outside a section.
+     * - Multi line comments are allowed, but the comment has to have the starting symbol in 
+     *   it's own line (\/\*) and the ending symbol also in it's own line (\*\/)
+     * 
+     * @remarks This parsing method asumes these rules, it doesn't validate them.
+     */
+    private static async parseCSSFile(filePath: string, globalVariables: PropertySection = {}): Promise<ThemeDefinition> {
+        const colorValues = {};
+        const rl = createInterface({ input: createReadStream(filePath) });
+
+        let insideComment = false;
+        let currentSection = '';
+        for await (const line of rl) {
+            const sectionMatch = line.match(CSS_SECTION_MATCH);
+            if (!insideComment && sectionMatch) {
+                currentSection = sectionMatch[0]
+                    .replace(/\/\*/, '') // Remove start of comment
+                    .replace(/\*\//, '') // Remove end of comment
+                    .trim()
+                    .replace(/([-_ ][a-z])/gi, (g) => 
+                        g.toUpperCase()
+                        .replace(/[-_ ]/, '')) // Make CammelCase
+                    .replace(/^./g, (g) => 
+                        g.toLowerCase()); // Ensure first char is lowercase
```

Comment:
why are using the camelCase convention? I think on the renderer we are using kebab-case

Reply:
camelCase is used just to get the section name in a the convention we use for typescript variables (which is camelCase), this to be able to properly generate the `CustomTheme` type.